sunpy/io/srs.py

```diff
@@ -0,0 +1,124 @@
+"""
+This module implements SRS File Reader.
+"""
+__author__ = "Sudarshan Konge"
+__email__ = "sudk1896@gmail.com"
+
+from astropy.table import Table, Column, vstack
+import collections, calendar, re
+
+__all__ = ['read']
+
+def read(filepath):
+    """
+    Method for reading an SRS File.
+    """
+    File = open(filepath, 'r')
+    lines = list()
+    for line in File:
+        arr = line.split()
+        store = list()
+        for i in range(0, min(len(arr), 8)):
+            store.append(arr[i])
+        lines.append(store)
+
+    #Store table meta-data. Only first two lines
+    #of SRS text file store the meta-data.
+    meta_data = collections.OrderedDict()
+    regex = re.compile('[^a-zA-Z]')
+    try:
+        meta_data[regex.sub('', lines[0][0])] = lines[0][1]
+        meta_data[regex.sub('', lines[1][0])] = ' '.join(lines[1][1:])
+    except IndexError:
+        print ("SRS meta-data is not available.")
+
+    #Problem: An array of strings. We need to extract
+    #three tables from these I, IA and II.
+
+    table = list() #This table holds the three tables which we would later merge.
+    indices = list()
+    
+    for i in range(0, len(lines)):
+        if (lines[i][0][0] == 'I'):
+            indices.append(i)
+    indices.append(len(lines))
+    indices.sort()
+    
+    for i in range(0, len(indices) - 1):
+        cols = lines[indices[i]+1]
+        temp_table = Table(names = cols, dtype=['object_']*len(cols))
+        #If a table is empty, we are not adding it.
```

Comment:
The problem with that is, essentially, whenever we process a column, we also want the datatype of that column. Initially, I am parsing all the columns (or rows, doesn't matter) as strings (in numpy that is **object_**. Once I get all my columns, I am converting the ones which can be converted to their respective datatypes. Example, **Nmbr** should be of type **float/int**. We do the same thing for other columns as well. Well, when a table has no values for that day, then all the columns have no values as well. So it would have **None** as the value. Lets say the **Nmbr** is None for a particular day, problem** is, **None ** is not of type **float/int**, so if we declare a column (in this case, **Nmbr**) as of type float, then we have to put a value of type **float**. And **None** is not that.